pkgs/development/python-modules/cron-descriptor/default.nix

```diff
@@ -1,32 +1,34 @@
 {
   lib,
-  python,
-  buildPythonPackage,
   fetchFromGitHub,
+  buildPythonPackage,
+  unittestCheckHook,
   mock,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "cron_descriptor";
-  version = "1.4";
+  version = "1.4.3";
+  pyproject = true;
+  build-system = [ setuptools ];
```

Comment:
but not this

Reply:
Normally we would write build-system between postPatch and nativeCheckInputs

Reply:
Is there a canonical ordering of those attributes defined somewhere?

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-function-buildpythonpackage-function seems to match what you described here, but isn't exhaustive. It can't be the argument order in mk-python-derivation.nix, I believe?

It be nice to avoid too much back and and forth on style-issues which don't make a semantic difference; especially for smaller changes like this. So while I am just adding nixpgks-rfc-style to a hook for my nixpkgs, is there either a list or even a linting option for this kind of issues?

Reply:
The order is pname, version, src, patchPhase, configurePhase, buildPhase, installPhase, fixupPhase, checkPhase, pythonImportsCheck, passthru, meta.

Reply:
But we seem to bike shed over those kind of things - see this thread among anothers. 

So if it's something that's important enough to request changes in otherwise okay PRs it should at least be written down at a linkable location (ideal would be linting later on). 

But otherwise this is very prone to cargo cult different bits of arcane knowledge among more regular contributors while making it close to impossible for newcomers to get this right without manual back and forth, just by reading docs & contribution guidelines